lib/h3core.js

```diff
@@ -251,9 +251,12 @@ function destroyGeoPolygon(geoPolygon) {
     C._free(C.getValue(geoPolygon + geofenceOffset + geofenceArrayOffset, 'i8*'));
     // Free the vertex array for the holes, if any
     const numHoles = C.getValue(geoPolygon + numHolesOffset, 'i32');
-    const holes = C.getValue(geoPolygon + holesOffset, 'i32');
-    for (let i = 0; i < numHoles; i++) {
-        C._free(C.getValue(holes + SZ_GEOFENCE * i + geofenceArrayOffset, 'i8*'));
+    if (numHoles > 0) {
+        const holes = C.getValue(geoPolygon + holesOffset, 'i32');
+        for (let i = 0; i < numHoles; i++) {
+            C._free(C.getValue(holes + SZ_GEOFENCE * i + geofenceArrayOffset, 'i8*'));
+        }
+        C._free(holes);
```

Comment:
Don't you need to go into each `hole` struct and free the inner coordinates array, too?

Reply:
That's what L257 does - gets the array pointer for each hole and frees it

Reply:
Bleh, I clearly shouldn't be reviewing C-memory-management-in-JS code while also working on my own code. Somehow I thought the loop was the outer array being deallocated.